generateComponentsTypes.ts

```diff
@@ -0,0 +1,18 @@
+import * as fs from 'fs';
+import { compile, compileFromFile } from 'json-schema-to-typescript';
+import { componentsDefinitionSchema_v1_0_x } from './lib/components-schema-v1_0_x';
+
+async function compileSchema(componentsSchema: object, outPath: string, schemaName: string) {
+    const ts = await compile(componentsSchema, schemaName, {
+        bannerComment: `/* tslint:disable */
```

Comment:
wonder why tslint is disabled?

Reply:
The auto-generated type definition conflicts with various rules, in particular
https://palantir.github.io/tslint/rules/array-type/